### ISSUE TITLE
Syntax Error in property assignment not caught till runtime

### ISSUE THREAD

**Issue by dmrvb** [twinbasic/twinbasic#1366]
**Describe the bug**
Label1.caption -= "aaa"     edits/compiles ok but gives runtime error 13 (type missmatch)

Note the "-" that my clumsy fingers inserted before the "="

**To Reproduce**
as above

**Expected behavior**
I think this is a bug as this should be caught at edit time?????

**Comment by wqweto** [comment]
JFYI, `-=` is a valid operator in TB e.g. `Dim Value As Long = 10 :  Value -= 4` brings `Value` to 6.

Other assignment operators include `+=` for strings too e.g. `Dim Caption As String = "Hello ": Caption += "World!"`

**Comment by dmrvb** [comment]
Yes, I did note that tB now has these C features, I have not used them yet but I will, they are good.
I think the question is, should the tB editor catch obvious type missmatch errors, or should they just give runtime errors (like VB) ? 
in your example above, I propose that we really should use the & for strings, and maybe give a warning on +.
This really is not a high priority issue, tB is now getting so close to compiling my code.

**Comment by wqweto** [comment]
These are not C/C++ features anymore but every non-VBA language has these: rust, go, Java, Kotlin, etc.

**Comment by dmrvb** [comment]
This could be an entertaining discussion for another time and place, but most languages now appear to use some variant of the C "Curly Brace" syntax even though this is most suited to mid/low level languages. VB and Python (and tB) are in a minority by using the more English "wordy" syntax, but I much prefer it. In my University days I was taught that one day we will program computers using ordinary English..... but then C took us in a direction.

**Comment by FullValueRider** [comment]
You should try nim.  Its a really really neat language.  If I put my tongue in my cheek I could say it is possibly the best preprocessor for C/C++ you can find.

**Comment by wqweto** [comment]
> VB and Python (and tB) are in a minority by using the more English "wordy" syntax, but I much prefer it.

Python at least tries to not be so "wordy": https://flexiple.com/python/assignment-operators-in-python/
